### DISCUSSION TITLE
Improving the functionality of Edge Functions

### DISCUSSION THREAD

**Discussion by jaitaiwan** [supabase/supabase#7851] [Feature Requests]
Hi Team,

From what I can tell, supabase functions are deployed using the deno deploy platform and are placed behind the functions relay (https://github.com/supabase/functions-relay). The limitation behind being able to serve websites via edge functions seems to come from this relay preventing any request that isn't POST or OPTIONS.

Is there some reason why we can't add something to the cli to allow not using the relay for example? I have a feeling the issue is managing resourcing behind the scenes, but I thought I'd ask anyway.

What prompted this was the [following quote](https://deno.com/blog/fresh-is-stable):
> Because Fresh is so reliant on dynamic server side rendering, it is imperative that this is fast. This makes Fresh very well suited for running on the edge in runtimes like Deno Deploy, Netlify Edge Functions, or Supabase Edge Functions

Thanks

**Comment by kiwicopple** [comment]
Great question! There are two reasons why we don't allow hosting of websites (right now):

1. We previously allowed HTML pages on Supabase Storage. At one point someone hosted a phishing site, and the entire domain was blocked by Google, causing all sorts of problems. 
2. If we provide web hosting, then there will be a huge number of requests for all the other features that come with it: branching, preview deploys, etc.

We don't have the capacity to deal with either of these at the moment. Vercel & Netlify are already amazing - we'd prefer to collaborate than compete. There are a lot of problems we need to focus on which are completely unsolved  - things like database migrations and branching. So for now, we are pouring our efforts into unsolved problems. 

(we might revisit this in the future once we've made more progress with databases + custom domains)
